selfdrive/controls/lib/pid.py

```diff
@@ -60,27 +61,33 @@ def reset(self):
     self.saturated = False
     self.control = 0
 
-  def update(self, setpoint, measurement, speed=0.0, check_saturation=True, override=False, feedforward=0., deadzone=0., freeze_integrator=False):
+  def update(self, setpoint, measurement, last_output=None, speed=0.0, check_saturation=True, override=False, feedforward=0., deadzone=0., freeze_integrator=False):
     self.speed = speed
 
     error = float(apply_deadzone(setpoint - measurement, deadzone))
     self.p = error * self.k_p
     self.f = feedforward * self.k_f
 
-    if override:
-      self.i -= self.i_unwind_rate * float(np.sign(self.i))
+    # Clip integrator based on the last output value
+    if last_output is not None:
+      if not freeze_integrator:
+        i_bf = self.p + self.i + self.f - last_output
```

Comment:
bf? can we give this a name that makes it's function clear?

Reply:
Can we make it clear self.i refers to the previous iteration and self.p and self.f do not, by renaming the appropriately? Now the code looks similar to this below `control = self.p + self.i + self.f`,  but has a completely different meaning.

Reply:
i_bf was "integrator BackFeed" so probably not even an accurate description anymore. 

Maybe "actuator_difference"

Reply:
> Can we make it clear self.i refers to the previous iteration and self.p and self.f do not, by renaming the appropriately? Now the code looks similar to this below `control = self.p + self.i + self.f`, but has a completely different meaning.

Could save self.control_last = control before the return and use that instead of each component.